src/service/onesignal/index.ts

```diff
@@ -1,36 +1,39 @@
 import {OneSignal} from 'react-native-onesignal';
 
 import getFeatureLoggerInstance, {EFeatureLogFlag} from '../../utils/log/FeatureLog';
+import {Monitoring} from '../../libraries/monitoring/sentry';
 import {getSubscribeableTopic} from '../topics';
 
 type TopicTag = {
   name: string;
 };
 
 const {featLog} = getFeatureLoggerInstance(EFeatureLogFlag.oneSignalUtils);
+const SET_TAG_TIMEOUT = 10 * 1000;
 
 const rebuildAndSubscribeTags = async () => {
   try {
     const response = await getSubscribeableTopic();
     const {topics = [], history = []} = response?.data || {};
 
     return new Promise((resolve) => {
-      history?.map((historyItem: TopicTag) => {
+      history?.forEach((historyItem: TopicTag) => {
         OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
-        return null;
+        Monitoring.logActions('remove tag', `better_community_${historyItem?.name}`);
       });
 
       setTimeout(() => {
-        topics?.map((topic: TopicTag) => {
+        topics?.forEach((topic: TopicTag) => {
           OneSignal.User.addTag(`better_community_${topic?.name}`, 'true');
-          return null;
+          Monitoring.logActions('add tag', `better_community_${topic?.name}`);
         });
 
         resolve(null);
-      }, 2000);
+      }, SET_TAG_TIMEOUT);
     });
   } catch (e) {
     console.log('error rebuilding and subscribing tags ', e?.message);
+    Monitoring.logError('error rebuilding and subscribing tags ', e);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `rebuildAndSubscribeTags` function could be improved. Currently, if an error occurs, it is logged to the console and sent to Sentry, but the Promise returned by the function will never resolve or reject. This could lead to unexpected behavior if other parts of your code are waiting for this Promise to settle. To fix this, you should reject the Promise in the catch block.

```diff
     console.log('error rebuilding and subscribing tags ', e?.message);
    Monitoring.logError('error rebuilding and subscribing tags ', e);
-    return null;
+    throw e;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->